### PR TITLE
 RTPS payload pool

### PR DESCRIPTION
**Pull request by MiguelCompany**
* Refs 9544. Adding new interfaces to API-ref

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

* Refs 9544. alias for references to API-ref

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

* Refs 9544. Update RTPS doc with references to API-ref

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

* Refs 9544. Documentation of Payload pool

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

* Refs 9544. Example with custom Payload pool

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

* Refs 9544. Suggestions and documentation check errors

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/643/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/648/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/649/)

File: docs/fastdds/api_reference/rtps/history/history.rst [hunk 1]
```diff
@@ -8,5 +8,7 @@ History
     /fastdds/api_reference/rtps/history/History_class
     /fastdds/api_reference/rtps/history/ReaderHistory
     /fastdds/api_reference/rtps/history/WriterHistory
+    /fastdds/api_reference/rtps/history/IChangePool
+    /fastdds/api_reference/rtps/history/IPayloadPool
```
Comment (EduPonz) [inline docs/fastdds/api_reference/rtps/history/history.rst:12]: Alphabetical order

File: docs/03-exports/aliases-api.include [hunk 2]
```diff
@@ -687,3 +716,4 @@
 .. |Subscriber::set_listener-api| replace:: :cpp:func:`Subscriber::set_listener()<eprosima::fastdds::dds::Subscriber::set_listener>`
 .. |DataReader::set_listener-api| replace:: :cpp:func:`DataReader::set_listener()<eprosima::fastdds::dds::DataReader::set_listener>`
 .. |Topic::set_listener-api| replace:: :cpp:func:`Topic::set_listener()<eprosima::fastdds::dds::Topic::set_listener>`
+
```
Comment (EduPonz) [inline docs/03-exports/aliases-api.include:719]:
```suggestion
```

File: code/CodeTester.cpp [hunk 3]
```diff
@@ -933,16 +933,115 @@ void rtps_api_example_create_entities()
     }
 
     //RTPS_API_WRITE_SAMPLE
-    //Request a change from the history
+    //Request a change from the writer
```
Comment (EduPonz) [inline code/CodeTester.cpp:936]:
```suggestion
    // Request a change from the writer
```

File: code/CodeTester.cpp [hunk 4]
```diff
@@ -933,16 +933,115 @@ void rtps_api_example_create_entities()
...
     CacheChange_t* change = writer->new_change([]() -> uint32_t {
         return 255;
     }, ALIVE);
     //Write serialized data into the change
     change->serializedPayload.length = sprintf((char*) change->serializedPayload.data, "My example string %d", 2) + 1;
-    //Insert change back into the history. The Writer takes care of the rest.
+    //Insert change into the history. The Writer takes care of the rest.
```
Comment (EduPonz) [inline code/CodeTester.cpp:942]:
```suggestion
    // Insert change into the history. The Writer takes care of the rest.
```

File: code/CodeTester.cpp [hunk 5]
```diff
@@ -933,16 +933,115 @@ void rtps_api_example_create_entities()
...
+            // Tell the CacheChange who needs to release its payload
+            cache_change.payload_owner(this);
+            
+            return true;
+        }
+
+        bool release_payload(
+                CacheChange_t& cache_change) override
+        {
+            // ensure precondition
```
Comment (EduPonz) [inline code/CodeTester.cpp:1001]:
```suggestion
            // Ensure precondition
```

File: code/CodeTester.cpp [hunk 6]
```diff
@@ -933,16 +933,115 @@ void rtps_api_example_create_entities()
...
+            assert (this == cache_change.payload_owner());
```
Comment (EduPonz) [inline code/CodeTester.cpp:1002]: Precondition only check in debug, I guess that's alright for the example
```suggestion
            assert(this == cache_change.payload_owner());
```

File: code/CodeTester.cpp [hunk 7]
```diff
@@ -933,16 +933,115 @@ void rtps_api_example_create_entities()
...
+    // Requesting a change to the Writer will provide one with an empty Payload taken from the pool
+    CacheChange_t* change = writer->new_change([]() -> uint32_t {
+        return 255;
+    }, ALIVE);
+
+    // Write serialized data into the change and add it to the history
+    change->serializedPayload.length = sprintf((char*) change->serializedPayload.data, "My example string %d", 2) + 1;
+    writer_history->add_change(change);
+
+
```
Comment (EduPonz) [inline code/CodeTester.cpp:1043]:
```suggestion
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/662/)

File: docs/fastdds/rtps_layer/rtps_layer.rst [hunk 8]
```diff
@@ -33,14 +37,15 @@ How to use the RTPS Layer
 We will now go over the use of the RTPS Layer like we did with the :ref:`dds_layer` one,
 explaining the new features it presents.
 
-We recommend you to look at the two examples of how to use this layer the distribution comes with while reading
-this section. They are located in `examples/RTPSTest_as_socket` and in `examples/RTPSTest_registered`
+We recommend you to look at the two examples describing how to use the RTPS layer that come with the distribution
+while reading this section.
+They are located in `examples/RTPSTest_as_socket` and in `examples/RTPSTest_registered`
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply docs/fastdds/rtps_layer/rtps_layer.rst]: Links have been added

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/664/)

**Review by EduPonz — APPROVED** [review]
LGTM :medal_sports:

**Pull request merged**